src/core/Microsoft.PowerToys.Settings.UI/ViewModels/GeneralViewModel.cs

```diff
@@ -2,14 +2,225 @@
+                if (_startup != value)
+                {
+                    RaisePropertyChanged();
+                }
+
+                _startup = value;
+            }
+        }
+
+        // Gets or sets a value indicating whether the powertoy elevated.
+        public bool IsElevated
+        {
+            get
+            {
+                return _isElevated;
+            }
+
+            set
+            {
+                if (_isElevated != value)
+                {
+                    RaisePropertyChanged();
+                }
+
+                _isElevated = value;
+            }
+        }
+
+        // Gets or sets a value indicating whether powertoys should run elevated.
+        public bool RunElevated
+        {
+            get
+            {
+                return _runElevated;
+            }
+
+            set
+            {
+                if (_runElevated != value)
+                {
+                    RaisePropertyChanged();
+                }
+
+                _runElevated = value;
+            }
+        }
+
+        public bool IsDarkThemeRadioButtonChecked
+        {
+            get
+            {
+                return _isDarkThemeRadioButtonChecked;
+            }
+
+            set
+            {
+                if (value == true)
+                {
+                    GeneralSettingsConfigs.theme = "dark";
+                    ShellPage.ShellHandler.RequestedTheme = ElementTheme.Dark;
+                    RaisePropertyChanged();
+                }
+
+                _isDarkThemeRadioButtonChecked = value;
+            }
+        }
+
+        public bool IsLightThemeRadioButtonChecked
+        {
+            get
+            {
+                return _isLightThemeRadioButtonChecked;
+            }
+
+            set
+            {
+                if (value == true)
+                {
+                    GeneralSettingsConfigs.theme = "light";
+                    ShellPage.ShellHandler.RequestedTheme = ElementTheme.Light;
+                    RaisePropertyChanged();
+                }
+
+                _isLightThemeRadioButtonChecked = value;
+            }
+        }
+
+        public bool IsSystemThemeRadioButtonChecked
+        {
+            get
+            {
+                return _isSystemThemeRadioButtonChecked;
+            }
+
+            set
+            {
+                if (value == true)
+                {
+                    GeneralSettingsConfigs.theme = "system";
+                    ShellPage.ShellHandler.RequestedTheme = ElementTheme.Default;
+                    RaisePropertyChanged();
+                }
+
+                _isSystemThemeRadioButtonChecked = value;
+            }
+        }
+
+        public void RaisePropertyChanged([CallerMemberName] string propertyName = null)
+        {
+            // Notify UI of property change
+            OnPropertyChanged(propertyName);
+
+            SettingsUtils.SaveSettings(GeneralSettingsConfigs.ToJsonString(), string.Empty);
```

Comment:
It can be separate PR/issue but we should not block UI thread for writing to file operation.

Reply:
Also, a more serious issue could be that IO exceptions are not being catched and handled.

Reply:
Saving these settings from the UI is not needed for one since the runner will do that once it has processed the settings. I have removed this line.